pkg/telemetry/telemetry.go

```diff
@@ -1,172 +1,18 @@
-//	timer.Record(ctx, elapsed, telemetry.StringAttr("endpoint", "/api"))
-//
-//	// Method 2: Automatic timing
-//	operation := timer.Start(ctx, telemetry.StringAttr("operation", "database_query"))
-//	defer operation.End()
-//
-// ## Histograms
-//
-// Histograms track the distribution of arbitrary values:
-//
-//	histogram, _ := tel.NewHistogram(telemetry.HistogramConfig{
-//	    Name:        "response_size",
-//	    Description: "Response size in bytes",
-//	    Unit:        "bytes",
-//	    Boundaries:  telemetry.SizeBoundaries,  // Predefined size boundaries
-//	})
-//	histogram.Record(ctx, responseSize, telemetry.StringAttr("endpoint", "/api"))
-//
-// # Complete HTTP Server Example
-//
-//	func main() {
-//	    ctx := context.Background()
-//
-//	    // Initialize telemetry
-//	    tel, err := telemetry.New(ctx, telemetry.Config{
-//	        ServiceName:    "my-http-server",
-//	        ServiceVersion: "1.0.0",
-//	        Environment:    "production",
-//	    })
-//	    if err != nil {
-//	        log.Fatal(err)
-//	    }
-//	    defer tel.Shutdown(ctx)
-//
-//	    // Create metrics
-//	    requestCounter, _ := tel.NewCounter(telemetry.CounterConfig{
-//	        Name:        "http_requests_total",
-//	        Description: "Total number of HTTP requests",
-//	    })
-//
-//	    requestTimer, _ := tel.NewTimer(telemetry.TimerConfig{
-//	        Name:        "http_request_duration_seconds",
-//	        Description: "HTTP request duration in seconds",
-//	    })
-//
-//	    // HTTP handler
-//	    http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-//	        start := time.Now()
-//
-//	        // Increment request counter
-//	        requestCounter.Add(r.Context(), 1,
-//	            telemetry.StringAttr("method", r.Method),
-//	            telemetry.StringAttr("path", r.URL.Path),
-//	        )
-//
-//	        // Your handler logic here
-//	        time.Sleep(100 * time.Millisecond) // Simulate work
-//	        w.Write([]byte("Hello, World!"))
-//
-//	        // Record request duration
-//	        elapsed := time.Since(start)
-//	        requestTimer.Record(r.Context(), elapsed,
-//	            telemetry.StringAttr("method", r.Method),
-//	            telemetry.StringAttr("path", r.URL.Path),
-//	            telemetry.IntAttr("status", 200),
-//	        )
-//	    })
-//
-//	    log.Println("Server starting on :8080")
-//	    log.Fatal(http.ListenAndServe(":8080", nil))
-//	}
-//
-// # Attributes
-//
-// Add contextual information to metrics using attributes:
-//
-//	telemetry.StringAttr("endpoint", "/api")
-//	telemetry.IntAttr("status_code", 200)
-//	telemetry.BoolAttr("success", true)
-//
-// # Predefined Boundaries
-//
-// The package provides predefined boundary values for common use cases:
-//
-//	telemetry.DefaultBoundaries  // General purpose boundaries
-//	telemetry.LatencyBoundaries  // For request latencies (seconds)
-//	telemetry.SizeBoundaries     // For response/request sizes (bytes)
```

Comment:
I agree some of these comments were a bit generic and described counters, gauges and such, but some others were specific to how this package is supposed to be used. I would've kept at least those, such as the options to use the package (component-specific telemetry instances vs. global instance) as well as the examples.

Reply:
My preference would be for dedicated comments on each telemetry method, I'd like to avoid blocking this merge on that, okay if we land in a follow on?